### ISSUE TITLE
Update xAxis in linePlusBarChart when changing domain

### ISSUE THREAD

**Issue by johnparn** [krispo/angular-nvd3#264]
Hi!

To start with - thanks for your effort making this library!

I'm using a linePlusBarChart to display data and the x-axis labels are not reflecting changes to the domain. For example I have data that is reported for the period 2015-09-25 until 2015-10-27. 

When setting the `xDomain` of `chart.bars` and `chart.lines` to another date interval, say 2015-08-01 and 2015-12-01 the data is properly rendered in the chart. However the dates of the xAxis are not updated, They still show the domain of the data, ie 2015-09-25 until 2015-10-27.

Using an ordinary linechart this works just fine. The data is shown properly and the xAxis labels are reflecting the changed dates.

Greatful for any input or hint how to solve this.

// John

chartOptions
```
{
  "chart": {
    "type": "linePlusBarChart",
    "bars": {
      "xDomain": [
        1441092055327,
        1451550055327
      ],
      "forceY": [
        0
      ],
      "padData": false,
      "interactive": true,
      "yDomain": [
        0,
        10
      ]
    },
    "lines": {
      "forceY": [
        0
      ],
      "xDomain": [
        1441092055327,
        1451550055327
      ],
      "interactive": true,
      "yDomain": [
        0,
        125
      ]
    },
    "legend": {
      "dispatch": {}
    },
    "focusEnable": false,
    "height": 360,
    "useInteractiveGuideline": true,
    "interactiveLayer": {
      "showGuideLine": true
    },
    "xDomain": [
      1441092055327,
      1451550055327
    ],
    "xAxis": {
      "showMaxMin": true,
      "rotateLabels": -45
    },
    "yAxis": {
      "tickValues": [
        0,
        10
      ]
    },
    "margin": {
      "bottom": 100
    },
    "legendLeftAxisHint": " (Vänster Y-axel)",
    "legendRightAxisHint": " (Höger Y-axel)",
    "transitionDuration": 250
  }
}
```

Data
```
[
  {
    "key": "Value (Vänster Y-axel)",
    "values": [
      {
        "x": 1443181994000,
        "y": "5"
      },
      {
        "x": 1443355627000,
        "y": "10"
      },
      {
        "x": 1443442096000,
        "y": "5"
      },
      {
        "x": 1443616371000,
        "y": "5"
      },
      {
        "x": 1443702783000,
        "y": "0"
      },
      {
        "x": 1443875590000,
        "y": "5"
      },
      {
        "x": 1444029466000,
        "y": "5"
      },
      {
        "x": 1444117308000,
        "y": "10"
      },
      {
        "x": 1444465080000,
        "y": "5"
      },
      {
        "x": 1444559443000,
        "y": "10"
      },
      {
        "x": 1444646058000,
        "y": "5"
      },
      {
        "x": 1444732597000,
        "y": "5"
      },
      {
        "x": 1444821334000,
        "y": "10"
      },
      {
        "x": 1444911032000,
        "y": "5"
      },
      {
        "x": 1445170276000,
        "y": "5"
      },
      {
        "x": 1445256986000,
        "y": "10"
      },
      {
        "x": 1445349301000,
        "y": "5"
      },
      {
        "x": 1445695782000,
        "y": "5"
      },
      {
        "x": 1445872639000,
        "y": "5"
      },
      {
        "x": 1445929665000,
        "y": "5"
      },
      {
        "x": 1445930754000,
        "y": "5"
      }
    ],
    "bar": true,
    "originalKey": "Value"
  },
  {
    "key": "Total (Höger Y-axel)",
    "values": [
      {
        "x": 1443181994002,
        "y": "5",
        "series": 0
      },
      {
        "x": 1443355627002,
        "y": "15",
        "series": 0
      },
      {
        "x": 1443442096002,
        "y": "20",
        "series": 0
      },
      {
        "x": 1443616371002,
        "y": "25",
        "series": 0
      },
      {
        "x": 1443702783002,
        "y": "25",
        "series": 0
      },
      {
        "x": 1443875590002,
        "y": "30",
        "series": 0
      },
      {
        "x": 1444029466002,
        "y": "35",
        "series": 0
      },
      {
        "x": 1444117308002,
        "y": "45",
        "series": 0
      },
      {
        "x": 1444465080002,
        "y": "50",
        "series": 0
      },
      {
        "x": 1444559443002,
        "y": "60",
        "series": 0
      },
      {
        "x": 1444646058002,
        "y": "65",
        "series": 0
      },
      {
        "x": 1444732597002,
        "y": "70",
        "series": 0
      },
      {
        "x": 1444821334002,
        "y": "80",
        "series": 0
      },
      {
        "x": 1444911032002,
        "y": "85",
        "series": 0
      },
      {
        "x": 1445170276002,
        "y": "90",
        "series": 0
      },
      {
        "x": 1445256986002,
        "y": "100",
        "series": 0
      },
      {
        "x": 1445349301002,
        "y": "105",
        "series": 0
      },
      {
        "x": 1445695782002,
        "y": "110",
        "series": 0,
        "color": "#aec7e8"
      },
      {
        "x": 1445872639002,
        "y": "115",
        "series": 0
      },
      {
        "x": 1445929665002,
        "y": "120",
        "series": 0
      },
      {
        "x": 1445930754002,
        "y": "125",
        "series": 0
      }
    ],
    "bar": false,
    "originalKey": "Total"
  }
]
```

**Comment by johnparn** [comment]
Here is an image depicting the problem. The date interval to show is selected using the datepickrs on the top of the image. The data is rendered in the right place and the charts are actually showing the correct date interval, but the axis labels are wrong in the linePlusBarChart.

![lineplusbarchart](https://cloud.githubusercontent.com/assets/1776929/10863310/29c439da-7fc9-11e5-8c1f-53a9ffd551b3.png)

**Comment by johnparn** [comment]
I found the issue to be located in the nvd3 library. `xDomain` was never honoured in the `nv.models.linePlusBarChart` function. I had to pull in `xDomain` into `chart._options` by adding the following row:

```xDomain: {get: function(){return xDomain;}, set: function(_){xDomain=_;}},```

Then I added xDomain variable to the domain method a bit further up:

```x2  .domain(xDomain || d3.extent(d3.merge(series1.concat(series2)), function(d) { return d.x } ))```

I also had to add `xDomain` as a public variable at the beginning of the function. 

Now the x-axis tick labels are changed along with the data.
